File: backend/app/schemas/session.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class SessionResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    session_type: str
    session_mode: str
    duration_seconds: int
    cefr_level_at_time: str | None
    started_at: datetime
    ended_at: datetime | None


class TranscriptResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    speaker: str
    content: str
    timestamp_offset_ms: int
    created_at: datetime


class ChatMessageCreate(BaseModel):
    content: str


class ChatMessageResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    sender: str
    content: str
    is_read: bool
    created_at: datetime

File: mobile/lib/models/session.dart
class LearningSession {
  final String id;
  final String sessionType;
  final String sessionMode;
  final int durationSeconds;
  final String? cefrLevelAtTime;
  final String startedAt;
  final String? endedAt;

  const LearningSession({
    required this.id,
    required this.sessionType,
    required this.sessionMode,
    required this.durationSeconds,
    this.cefrLevelAtTime,
    required this.startedAt,
    this.endedAt,
  });

  factory LearningSession.fromJson(Map<String, dynamic> json) {
    return LearningSession(
      id: json['id'] as String,
      sessionType: json['session_type'] as String,
      sessionMode: json['session_mode'] as String,
      durationSeconds: json['duration_seconds'] as int,
      cefrLevelAtTime: json['cefr_level_at_time'] as String?,
      startedAt: json['started_at'] as String,
      endedAt: json['ended_at'] as String?,
    );
  }
}
